### PR TITLE
Models should be records, drop T4 transformation

### PR DESCRIPTION
**Pull request by emanuelgaspar**

### CODE REVIEW

File: src/Trakx.Fireblocks.ApiClient/Registration/DependencyInjection.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Trakx.Fireblocks.ApiClient;
 /// <summary>
 /// Extension methods for adding the Fireblocks API client to the service collection.
 /// </summary>
-public static partial class AddFireblocksClientExtension
+public static partial class DependencyInjection
```
Comment (emanuelgaspar) [inline src/Trakx.Fireblocks.ApiClient/Registration/DependencyInjection.cs:13]: Happy to revert, but in the api clients, there's already so many files called "XxxApiClientxxx" that using our standard `DependencyInjection` is more intuitive.

File: src/Trakx.Fireblocks.ApiClient/nswag.json [hunk 2]
```diff
@@ -9,92 +9,99 @@
...
+
       "generateClientClasses": true,
       "generateClientInterfaces": true,
+      "clientBaseClass": "AuthorisedClient",
       "clientBaseInterface": "IFireblocksApiClientBase",
-      "injectHttpClient": false,
-      "disposeHttpClient": true,
-      "protectedMethods": [],
-      "generateExceptionClasses": false,
-      "exceptionClass": "ApiException",
```
Comment (emanuelgaspar) [inline src/Trakx.Fireblocks.ApiClient/nswag.json:21]: I've been reorganizing nswag settings to block them together in "themes", with the most relevant stuff at the top. So this looks like a lot of changes, but it's just moving lines around.

File: src/Trakx.Fireblocks.ApiClient/nswag.json [hunk 3]
```diff
@@ -9,92 +9,99 @@
...
-      "parameterDateTimeFormat": "yyyy-MM-ddTHH:mm:ssZ",
-      "parameterDateFormat": "yyyy-MM-dd",
-      "generateUpdateJsonSerializerSettingsMethod": true,
-      "useRequestAndResponseSerializationSettings": false,
-      "serializeTypeInformation": false,
-      "queryNullValue": "null",
-      "className": "{controller}Client",
+      "configurationClass": "IClientConfigurator",
       "operationGenerationMode": "MultipleClientsFromFirstTagAndOperationId",
+      "generatePrepareRequestAndProcessResponseAsAsyncMethods": true,
```
Comment (emanuelgaspar) [inline src/Trakx.Fireblocks.ApiClient/nswag.json:19]: This is the setting that allows us to drop the T4.

File: tests/Trakx.Fireblocks.ApiClient.Tests/Unit/FireblocksApiClientPartialsTests.cs
Comment (emanuelgaspar) [inline tests/Trakx.Fireblocks.ApiClient.Tests/Unit/FireblocksApiClientPartialsTests.cs:1]: We no longer need to check this, as we don't have partials.

File: tests/Trakx.Fireblocks.ApiClient.Tests/Unit/GeneratedCodeTests.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+namespace Trakx.Fireblocks.ApiClient.Tests.Unit;
+
+public class GeneratedCodeTests
+{
+    [Fact]
+    public void Models_should_be_records()
+    {
+        var content = File.ReadAllText(ProjectPaths.ModelsFile);
+        content.Should().Contain("public partial record");
```
Comment (emanuelgaspar) [inline tests/Trakx.Fireblocks.ApiClient.Tests/Unit/GeneratedCodeTests.cs:9]: This fixes the issue found here:

https://github.com/trakx/custodian/pull/110/files
image

**Review by joaofbvieira — APPROVED** [review]

**Pull request merged**